### DISCUSSION TITLE
Shader Canvas Tool

### DISCUSSION THREAD

**Discussion by AMZN-Liv** [o3de/rfcs#25] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, creating shaders requires programming knowledge of the shader language AZSL. Shaders are an important component of visual authoring for systems in-engine, and editing visual components with a text-base language is challenging. A visual editor for modifying shaders would reduce the barrier of entry for creating new shaders, and empower a larger audience to create shaders in the editor.

**Describe the solution you'd like**
A shader canvas visual editing tool, potentially similar to Script Canvas or the Material editor

**Comment by arvinmoses** [comment]
I'm a material tech artist.  I'd very much love something like this, though I don't have a big aversion to coding in text.  I think possibly more so I'd be more concerned about ensuring a good structure for exposing shader controls, easy ways to add new features, and tools to hide and show various parameters for the artists.

A dynamic layer system is also very important to a modern material pipeline.

One thing to possibly look into is easy translation from a materialx shader and working together with Adobe Substance to do this.  I can attempt to help arrange if desired
